sound/soc/sof/ipc4-topology.c

```diff
@@ -1050,16 +1050,21 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		return -EINVAL;
 	}
 
-	/* copy input format */
-	memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
-	       sizeof(struct sof_ipc4_audio_format));
-
-	/* set base_cfg ibs/obs */
-	base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
-	base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
+	/* copy input format and set base_cfg ibs*/
+	if (available_fmt->input_pin_fmts) {
+		memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
+		       sizeof(struct sof_ipc4_audio_format));
+		base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
+
+		dev_dbg(sdev->dev, "init input audio formats for %s\n", swidget->widget->name);
+		sof_ipc4_dbg_audio_format(sdev->dev, &available_fmt->input_pin_fmts[i], 1);
+	} else {
+		dev_dbg(sdev->dev, "No init input audio formats for %s\n", swidget->widget->name);
```

Comment:
this is printed when no input pin is defined, sure it means that there is no input format, but...